File: src/com/potter/riderunity3dconnector/SettingsDialog.java
package com.potter.riderunity3dconnector;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.awt.*;
import java.text.NumberFormat;
import java.util.Locale;

public class SettingsDialog extends DialogWrapper {

    private Project project;

    protected SettingsDialog(@Nullable Project project, boolean canBeParent) {
        super(project, canBeParent);
        this.project = project;
        init();
        setTitle("Unity3D Connector Status");
    }

    @Nullable
    @Override
    protected JComponent createCenterPanel() {

        return createSettingsPanel();
    }

    private JPanel createSettingsPanel() {
        JPanel mainPanel = new JPanel(new BorderLayout());
        JPanel fieldsPanel = new JPanel(new GridLayout(2,2));

        JLabel startServerLabel = new JLabel();
        startServerLabel.setText("Listening Unity: ");

        JCheckBox cb = new JCheckBox();
        cb.setSelected(UnityConnectorServer.getInstance().isRunning());
        cb.addActionListener(e -> {
            JCheckBox cb1 = (JCheckBox)e.getSource();
            if(cb1.isSelected()) {
                boolean started = UnityConnectorServer.getInstance().startServer();
                cb1.setSelected(started);
            } else {
                UnityConnectorServer.getInstance().stopServer();
            }
        });

        fieldsPanel.add(startServerLabel);
        fieldsPanel.add(cb);

        mainPanel.add(fieldsPanel, BorderLayout.CENTER);
        return mainPanel;
    }

    public void stopListening() {
        UnityConnectorServer.getInstance().stopServer();
    }
}

File: src/com/potter/riderunity3dconnector/SettingsDialog.kt
package com.potter.riderunity3dconnector

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import java.awt.BorderLayout
import java.awt.GridLayout
import javax.swing.JCheckBox
import javax.swing.JComponent
import javax.swing.JLabel
import javax.swing.JPanel

class SettingsDialog(project: Project?, canBeParent: Boolean) : DialogWrapper(project, canBeParent) {

    init {
        init()
        title = "Unity3D Connector Status"
    }

    override fun createCenterPanel(): JComponent? {

        return createSettingsPanel()
    }

    private fun createSettingsPanel(): JPanel {
        val mainPanel = JPanel(BorderLayout())
        val fieldsPanel = JPanel(GridLayout(2, 2))

        val startServerLabel = JLabel()
        startServerLabel.text = "Listening Unity: "

        val cb = JCheckBox()
        cb.isSelected = UnityConnectorServer.instance.isRunning
        cb.addActionListener { e ->
            val cb1 = e.source as JCheckBox
            if (cb1.isSelected) {
                val started = UnityConnectorServer.instance.startServer()
                cb1.isSelected = started
            } else {
                UnityConnectorServer.instance.stopServer()
            }
        }

        fieldsPanel.add(startServerLabel)
        fieldsPanel.add(cb)

        mainPanel.add(fieldsPanel, BorderLayout.CENTER)
        return mainPanel
    }

    fun stopListening() {
        UnityConnectorServer.instance.stopServer()
    }
}
